### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by analogic**
Made modem.go compatible with siemens MC75, extended methods to allow future SMS reading (will do in couple days).  Added optional http basic auth. (all changes should not cause any problems to current installations)

### CODE REVIEW

File: dashboard/main.go [hunk 1]
```diff
@@ -62,7 +65,7 @@ func main() {
 	gosms.InitWorker(modems, bufferSize, bufferLow, loaderTimeout, loaderCountout, loaderTimeoutLong)
 
 	log.Println("main: Initializing server")
-	err = InitServer(serverhost, serverport)
+	err = InitServer(serverhost, serverport, serverusername, serverpassword)
```
Comment (Omie) [inline dashboard/main.go]: sorry, not in touch with Golang off late. What happens if username and passwords are empty? Do you still get a prompt and requires to hit enter?

**Comment by Omie** [comment]
hey @analogic, thanks for the pr. Neither of me and @madhurjain has bandwidth to maintain this project. I will merge the changes as long as they look good enough just after reading the code :)
Could you please add contributors.txt and your name as contributor in it? You may just update the readme with the same if you feel like!

File: dashboard/main.go (hunk 1 above)
Comment (analogic) [inline dashboard/main.go]: Nothing, web server will be open as default (conf.ini has USERNAME and PASSWORD commented out)

**Comment by analogic** [comment]
I will add credits next time since I need catch incomming sms also. So I will definitely continue to work on this project :)

**Comment by Omie** [comment]
that would be another PR, right? should I merge and close this one?

**Comment by analogic** [comment]
Yep, it is up to you but sure you can. :)

**Pull request merged**